backend/src/auth/auth.service.ts

```diff
@@ -12,16 +11,17 @@ export class AuthService {
   ) {}
 
   async login(userDto: LoginUserDto) {
-    const userList = await this.usersService.findAll({
-      username: userDto.username,
-    });
-    if (userList.data.length !== 1) {
-      throw new EntityDoesNotExistError(`User: ${userDto.username}`);
+    try {
+      const user = await this.usersService.findByName(userDto.username);
+      const payload = { id: user.id, username: user.username };
+      return {
+        access_token: this.jwtService.sign(payload),
+      };
+    } catch (EntityNotFoundError) {
+      return {
+        status: 401,
+        message: 'Username is incorrect.',
+      };
     }
```

Comment:
This works well, but it might be cleaner to move this catch block into the login function of `auth.controller.ts`. The idea being that the service carries out the operations but the controller catches the `EntityNotFoundError` and decides what HTTP response to send to the client.

Another advantage is that the controller can use Nest's [Built-in HTTP exceptions](https://docs.nestjs.com/exception-filters#built-in-http-exceptions) to avoid manually creating the response
```suggestion
    } catch (EntityNotFoundError) {
      throw new UnauthorizedException('Username is incorrect');
    }
```

This will automatically create a similar object to send to the client

Reply:
Yes, I think it will be better not to return response JSON in a service, as my understanding is that a service is supposed to be usable application-wide. A controller exposes functionality from services to the outside world.